AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1155,6 +1160,33 @@ - (void)calculatedLayoutDidChange
   // subclass override
 }
 
+- (void)setMeasurementOptions:(ASDisplayNodePerformanceMeasurementOptions)measurementOptions
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  _measurementOptions = measurementOptions;
+}
+
+- (ASDisplayNodePerformanceMeasurementOptions)measurementOptions
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _measurementOptions;
+}
+
+- (NSDictionary *)performanceMeasurements
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  NSMutableDictionary *measurements = [NSMutableDictionary dictionaryWithCapacity:4];
+  if (_measurementOptions & ASDisplayNodePerformanceMeasurementOptionLayoutSpec) {
+    measurements[ASDisplayNodeLayoutSpecTotalTimeKey] = @(_layoutSpecTotalTime);
```

Comment:
@shamanskyh the design of this is efficient insofar as at recording time, cost is very low.  Accessing this property is pretty costly though, as the numbers also have to be boxed to put in the dictionary (and then accessed from it).

Would it be reasonable to define a struct type that all of these numbers can be contained within, and use that both as the instance variable storage and directly returned from this method?  If we want easy printing, maybe a good balance would be a basic NSObject subclass with simple @properties for each of these.

This has the downside (which you probably already considered) of making it more ambiguous which values are set vs zero.  It might be a nastier API, but we could use NaN to represent those defaults.